src/search/engines/bingimages/bingimages.go

```diff
@@ -0,0 +1,222 @@
+
+	var pageRankCounter []int = make([]int, options.MaxPages*Info.ResultsPerPage)
+
+	col.OnHTML(dompaths.Result, func(e *colly.HTMLElement) {
+		dom := e.DOM
+
+		var jsonMetadata JsonMetadata
+		metadataS, metadataExists := dom.Find(dompaths.Metadata).Attr("m")
+		if !metadataExists {
+			log.Trace().
+				Str("engine", Info.Name.String()).
+				Msg("Matched result, but couldn't retrieve data")
+			return
+		}
+
+		if err := json.Unmarshal([]byte(metadataS), &jsonMetadata); err != nil {
+			log.Error().
+				Err(err).
+				Msg("bingimages.Search() -> onHTML: failed to unmarshal metadata")
+			return
+		}
+
+		if jsonMetadata.Purl == "" || jsonMetadata.Murl == "" || jsonMetadata.Turl == "" {
+			log.Error().
+				Str("engine", Info.Name.String()).
+				Str("jsonMetadata", metadataS).
+				Str("url", jsonMetadata.Purl).
+				Str("original", jsonMetadata.Murl).
+				Str("thumbnail", jsonMetadata.Turl).
+				Msg("bingimages.Search() -> onHTML: Couldn't find image URL")
+			return
+		}
+
+		titleText := strings.TrimSpace(dom.Find(dompaths.Title).Text())
+		if titleText == "" {
+			log.Error().
+				Str("engine", Info.Name.String()).
+				Str("jsonMetadata", metadataS).
+				Msg("bingimages.Search() -> onHTML: Couldn't find title")
+			return
+		}
+
+		// this returns "2000 x 1500 · jpeg"
+		imgFormatS := strings.TrimSpace(dom.Find(dompaths.ImgFormatStr).Text())
+		if imgFormatS == "" {
+			// this happens when bingimages returns video instead of image
+			log.Trace().
+				Str("engine", Info.Name.String()).
+				Str("jsonMetadata", metadataS).
+				Str("title", titleText).
+				Msg("bingimages.Search() -> onHTML: Couldn't find image format")
+			return
+		}
+
+		// convert to "2000x1500·jpeg"
+		imgFormatS = strings.ReplaceAll(imgFormatS, " ", "")
+		// remove everything after 2000x1500
+		imgFormatS = strings.Split(imgFormatS, "·")[0]
+		// create height and width
+		imgFormat := strings.Split(imgFormatS, "x")
+
+		imgH, err := strconv.Atoi(imgFormat[0])
+		if err != nil {
+			log.Error().
+				Err(err).
+				Str("engine", Info.Name.String()).
+				Str("height", imgFormat[0]).
+				Str("jsonMetadata", metadataS).
+				Str("title", titleText).
+				Str("imgFormatS", imgFormatS).
+				Msg("bingimages.Search() -> onHTML: Failed to convert original height to int")
+			return
+		}
+
+		imgW, err := strconv.Atoi(imgFormat[1])
+		if err != nil {
+			log.Error().
+				Err(err).
+				Str("engine", Info.Name.String()).
+				Str("width", imgFormat[1]).
+				Str("jsonMetadata", metadataS).
+				Str("title", titleText).
+				Str("imgFormatS", imgFormatS).
+				Msg("bingimages.Search() -> onHTML: Failed to convert original width to int")
+			return
+		}
+
+		thmbHS, thmbHSExists := dom.Find(dompaths.ThumbnailHeight).Attr("height")
+		if !thmbHSExists {
+			log.Error().
+				Str("engine", Info.Name.String()).
+				Str("jsonMetadata", metadataS).
+				Str("title", titleText).
+				Msg("bingimages.Search() -> onHTML: Couldn't find thumbnail height")
+			return
```

Comment:
If its expected that this happens (since videos are also matched and dont have this field?), it either, shouldn't error out, or greater care should be taken to ensure videos aren't matched.